Lib/sqlite3/test/userfunctions.py

```diff
@@ -275,6 +276,30 @@ def CheckAnyArguments(self):
         val = cur.fetchone()[0]
         self.assertEqual(val, 2)
 
+    @unittest.skipIf(sqlite.sqlite_version_info < (3, 8, 3), "deterministic parameter not supported")
+    def CheckFuncDeterministic(self):
+        mock = unittest.mock.Mock()
+        mock.return_value = None
+        query = "select deterministic() = deterministic()"
+
+        self.con.create_function("deterministic", 0, mock, deterministic=False)
+        self.con.execute(query)
+        self.assertEqual(mock.call_count, 2)
+
+        mock.reset_mock()
+        self.con.create_function("deterministic", 0, mock, deterministic=True)
+        self.con.execute(query)
+        self.assertEqual(mock.call_count, 1)
+
+    @unittest.skipIf(sqlite.sqlite_version_info >= (3, 8, 3), "SQLite < 3.8.3 needed")
+    def CheckFuncDeterministicNotSupported(self):
+        with self.assertRaises(sqlite.NotSupportedError):
+            self.con.create_function("deterministic", 0, int, deterministic=True)
```

Comment:
`self.con.create_function("deterministic", 0, int, deterministic=False)` should work even on SQLite < 3.8.3, right?

Reply:
Yes, just pushed the fix.